packages/iov-bns/src/bnsconnection.ts

```diff
@@ -623,18 +606,25 @@ export class BnsConnection implements BcpAtomicSwapConnection {
   }
 
   public async getFeeQuote(transaction: UnsignedTransaction): Promise<Fee> {
-    if (isBnsTx(transaction)) {
-      const firstToken = (await this.getAllTickers())[0];
-      return {
-        tokens: {
-          quantity: "0",
-          fractionalDigits: firstToken.fractionalDigits,
-          tokenTicker: firstToken.tokenTicker,
-        },
-      };
+    if (!isBnsTx(transaction)) {
+      throw new Error("Received transaction of unsupported kind.");
+    }
+    // get gconf fee
+    // TODO: get product fee for this type
+    // take maximum
+    const res = await this.query("/", Encoding.toAscii("gconf:cash:minimal_fee"));
```

Comment:
Why do we need to support the human readable token format in a machine-machine interface? I thought this was created for genesis editing only.

Reply:
Yeah, it was.
But somehow gconf doesn't actually parse the genesis file before writing to db (as it should), but rather copies it blindly until we read it.
Didn't realize this gotcha, til I tried parsing with protobuf Coin and it failed.

We should discuss gconf and improvements in a design session.